include/z64audio.h

```diff
@@ -0,0 +1,1208 @@
+    u32 wantSample;
+    u32 wantCustom;
+} AudioPoolSplit4; // size = 0x10
+
+typedef struct {
+    /* 0x00 */ u32 endAndMediumKey;
+    /* 0x04 */ SoundFontSample* sample;
+    /* 0x08 */ u8* ramAddr;
+    /* 0x0C */ u32 encodedInfo;
+    /* 0x10 */ s32 isFree;
+} AudioPreloadReq; // size = 0x14
+
+typedef struct {
+    union{
+        u32 opArgs;
+        struct {
+            u8 op;
+            u8 arg0;
+            u8 arg1;
+            u8 arg2;
+        };
+    };
+    union {
+        void* data;
+        f32 asFloat;
+        s32 asInt;
+        u16 asUShort;
+        s8 asSbyte;
+        u8 asUbyte;
+        u32 asUInt;
+    };
+} AudioCmd; // size = 0x8
+
+typedef struct {
+    /* 0x00 */ s8 status;
+    /* 0x01 */ s8 delay;
+    /* 0x02 */ s8 medium;
+    /* 0x04 */ u8* ramAddr;
+    /* 0x08 */ u32 curDevAddr;
+    /* 0x0C */ u8* curRamAddr;
+    /* 0x10 */ u32 bytesRemaining;
+    /* 0x14 */ u32 chunkSize;
+    /* 0x18 */ s32 unkMediumParam;
+    /* 0x1C */ u32 retMsg;
+    /* 0x20 */ OSMesgQueue* retQueue;
+    /* 0x24 */ OSMesgQueue msgQueue;
+    /* 0x3C */ OSMesg msg;
+    /* 0x40 */ OSIoMesg ioMesg;
+} AudioAsyncLoad; // size = 0x58
+
+typedef struct {
+    /* 0x00 */ u8 medium;
+    /* 0x01 */ u8 seqOrFontId;
+    /* 0x02 */ u16 instId;
+    /* 0x04 */ s32 unkMediumParam;
+    /* 0x08 */ s32 curDevAddr;
+    /* 0x0C */ u8* curRamAddr;
+    /* 0x10 */ u8* ramAddr;
+    /* 0x14 */ s32 status;
+    /* 0x18 */ s32 bytesRemaining;
+    /* 0x1C */ s8* isDone;
+    /* 0x20 */ SoundFontSample sample;
+    /* 0x30 */ OSMesgQueue msgqueue;
+    /* 0x48 */ OSMesg msg;
+    /* 0x4C */ OSIoMesg ioMesg;
+} AudioSlowLoad; // size = 0x64
+
+typedef struct {
+    /* 0x00 */ u32 romAddr;
+    /* 0x04 */ u32 size;
+    /* 0x08 */ s8 medium;
+    /* 0x09 */ s8 cachePolicy;
+    /* 0x0A */ s16 shortData1;
+    /* 0x0C */ s16 shortData2;
+    /* 0x0E */ s16 shortData3;
+} AudioTableEntry; // size = 0x10
+
+typedef struct {
+    /* 0x00 */ s16 numEntries;
+    /* 0x02 */ s16 unkMediumParam;
+    /* 0x04 */ u32 romAddr;
+    /* 0x08 */ char pad[0x8];
+    /* 0x10 */ AudioTableEntry entries[1]; // (dynamic size)
+} AudioTable; // size >= 0x20
+
+typedef struct {
+    /* 0x00 */ OSTask task;
+    /* 0x40 */ OSMesgQueue* taskQueue;
+    /* 0x44 */ void* unk_44; // probably a message that gets unused.
+    /* 0x48 */ char unk_48[0x8];
+} AudioTask; // size = 0x50
+
+typedef struct {
+    /* 0x00 */ u8* ramAddr;
+    /* 0x04 */ u32 devAddr;
+    /* 0x08 */ u16 sizeUnused;
+    /* 0x0A */ u16 size;
+    /* 0x0C */ u8 unused;
+    /* 0x0D */ u8 reuseIndex; // position in sSampleDmaReuseQueue1/2, if ttl == 0
+    /* 0x0E */ u8 ttl;        // duration after which the DMA can be discarded
+} SampleDma; // size = 0x10
+
+typedef struct {
+    /* ?0x00 */ s32 sampleBankId1;
+    /* ?0x04 */ s32 sampleBankId2;
+    /* ?0x08 */ s32 baseAddr1;
+    /* ?0x0C */ s32 baseAddr2;
+    /* ?0x10 */ u32 medium1;
+    /* ?0x14 */ u32 medium2;
+} RelocInfo; // size = 0x18
```

Comment:
What is this struct exactly?
I'm a bit worried "RelocInfo" can be confused with the overlay's relocs

Reply:
RelocInfo is a part of the file `audio_load`. Sound fonts for instruments/drums/sfx contain table of a bunch of sounds, and each of those sounds has a pointer to the raw audio sample data it uses. Originally those pointers are offsets, and `audio_load` "relocates" them by adding the sample bank base pointer. 

OoT puts this struct directly in `audio_load` instead of the audio header file, but I had a tendency to just put everything in `z64audio.h` until its worked out. 
https://github.com/zeldaret/oot/blob/80edcc3801723348c4c4350dfe8194a6274e5357/src/code/audio_load.c#L16-L23

Would this name still be an issue if I move this struct to `audio_load`? (i.e take it out here and bring it in during the audio_load PR)

Reply:
Actually, this is being moved to `audio_load` so I'll just take it out